server/controllers/handleLogin.js

```diff
@@ -11,19 +11,20 @@ const handleLogin = (req, res, next) => {
       if (data.rowCount === 1) {
         return getPassword(data.rows[0].email);
       } else {
-        throw customizedError("Email and password don't  match ");
+        throw customizedError("Email and password don't  match " , 400);
       }
     })
     .then((data) => {
       return compare(password, data.rows[0].password);
     })
     .then((isMatch) => {
       if (isMatch) {
-        return signPromise(email)
+        return getId(email)
```

Comment:
what if you can get the password from the previous query!
So we save an extra step here